contracts/Manager.sol

```diff
@@ -508,6 +661,79 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
     }
 
+    /**
+     * @notice Used to withdraw CELO from the system from specific validator group
+     * that account voted for previously.
+     * @param specificGroup The specific validator group that we want to withdraw from.
+     * @param withdrawal The amount of stCELO to withdraw.
+     * @return groups The groups to withdraw from.
+     * @return votes The amount to withdraw from each group.
+     */
+    function withdrawFromSpecificGroup(
+        address specificGroup,
+        uint256 withdrawal,
+        uint256 stCeloAmount
+    ) private returns (address[] memory groups, uint256[] memory votes) {
+        uint256 votesRemaining = account.getCeloForGroup(specificGroup);
+        uint256 votesToWithdraw = Math.min(votesRemaining, withdrawal);
+
+        groups = new address[](1);
+        votes = new uint256[](1);
+        groups[0] = specificGroup;
+        votes[0] = votesToWithdraw;
+
+        if (stCeloAmount > specificStrategyTotalStCeloVotes[specificGroup]) {
+            revert NotEnoughStCeloForSpecificGroup(specificGroup);
+        }
+
+        specificStrategyTotalStCeloVotes[specificGroup] -= stCeloAmount;
+
+        if (specificStrategyTotalStCeloVotes[specificGroup] == 0) {
+            if (activeGroups.contains(specificGroup)) {
+                if (specificStrategyVotedGroups.remove(specificGroup)) {
+                    delete specificStrategyTotalStCeloVotes[specificGroup];
+                    emit GroupRemoved(specificGroup);
+                }
+            } else {
+                // if group is being removed from specific strategy
+                // groups and is not active make sure to withdraw all
+                // possibly remaining funds from there (there might be some dust)
+                _deprecateGroup(specificGroup);
+            }
+        }
+    }
+
+    /**
+     * @notice Whenever stCELO is being transferd we will check whether origin and target
+     * account use same strategy. If strategy differs we will schedule votes for tranfer.
+     * @param from The from account.
+     * @param to The to account.
+     * @param amount The amount.
+     */
+    function transfer(
+        address from,
+        address to,
+        uint256 amount
+    ) public onlyStakedCelo {
+        address fromStrategy = strategies[from];
+        address toStrategy = strategies[to];
+
+        if (fromStrategy == toStrategy) {
+            // either both addresses use default strategy
+            // or both addresses use same specific strategy
+            return;
+        }
+
+        address[] memory fromGroups;
+        uint256[] memory fromVotes;
+        (fromGroups, fromVotes) = generateGroupsVotesToDistributeTo(amount, fromStrategy);
+
+        address[] memory toGroups;
+        uint256[] memory toVotes;
+        (toGroups, toVotes) = generateGroupsVotesToDistributeTo(amount, toStrategy);
+        account.scheduleTransfer(fromGroups, fromVotes, toGroups, toVotes);
+    }
```

Comment:
Where do we update `specificStrategyTotalStCeloVotes`?

Reply:
It is currently being updated in:

deposit
withdrawFromSpecificGroup
_deprecateGroup
_transfer